contracts/contracts/token/WOETH.sol

```diff
@@ -49,7 +80,161 @@ contract WOETH is ERC4626, Governable, Initializable {
         external
         onlyGovernor
     {
+        //@dev TODO: we could implement a feature where if anyone sends OETH direclty to
+        // the contract, that we can let the governor transfer the excess of the token.
         require(asset_ != address(asset()), "Cannot collect OETH");
         IERC20(asset_).safeTransfer(governor(), amount_);
     }
+
+    /**
+     * @dev This function converts requested OETH token amount to its underlying OETH
+     * credits value that is stored internally in OETH.sol and is required in order to
+     * be able to rebase.
+     *
+     * @param oethAmount Amount of OETH to be converted to OETH credits
+     * @return amount of OETH credits the OETH amount corresponds to
+     */
+    function _creditsPerAsset(uint256 oethAmount)
```

Comment:
nit: This function name confuses me a bit. When I see `per asset`, I assume it's gonna return credits for 1 OETH. But it takes in a `oethAmount` and multiplies the credits by it. So, it sounds more like `creditsForAssetAmount` than `creditsPerAsset`

Reply:
yup agreed: https://github.com/OriginProtocol/origin-dollar/pull/2106/commits/1a6a16fc4af1f115f5c84a331df821da4b551d47

Reply:
good point. I have also renamed assets & shares since in the context of WOETH those can only be OETH & WOETH: https://github.com/OriginProtocol/origin-dollar/pull/2106/commits/2223600a1d8b8bf23c940a7debe124ddb1e4f44d